proto/workflow.proto

```diff
@@ -169,8 +172,25 @@ message ExecuteWorkflowResponse {
   // The response context.
   context.ResponseContext response_context = 1;
 
+  // TODO(Maggie): Deprecate this field and clean up references
   // The invocation ID of the workflow execution.
   string invocation_id = 2;
+
+  // A list of the actions executed by the API.
+  repeated WorkflowAction actions_started = 3;
+}
+
+message WorkflowAction {
+  // Corresponds to an action in buildbuddy.yaml, or an action name in the
+  // default workflow config if buildbuddy.yaml is missing from the repo. Ex.
+  // "Test all targets"
+  string action_name = 1;
+
+  // The invocation ID of the execution of the action.
+  string invocation_id = 2;
+
+  // The status code for executing the action.
+  string grpc_status_code = 3;
```

Comment:
you can use `google.rpc.Status` here (from `google/rpc/status.proto`) so that the client can see the full error message in addition to the status code, which will help with debugging. There's a handy util `gstatus.Convert(err).Proto()` to get a status proto from a go error (also returns the correct `nil` status when `err == nil`)

Reply:
Ah cool!